tensorflow_addons/custom_ops/image/python/distort_image_ops.py

```diff
@@ -0,0 +1,143 @@
+        `lower_value`, or `upper_Value` is invalid.
+    """
+    if max_delta_hue < 0:
+        raise ValueError("max_delta must be non-negative.")
+
+    if lower_saturation < 0:
+        raise ValueError("lower_saturation must be non-negative.")
+
+    if lower_value < 0:
+        raise ValueError("lower_value must be non-negative.")
+
+    if lower_saturation > upper_saturation:
+        raise ValueError("lower_saturation must be < upper_saturation.")
+
+    if lower_value > upper_value:
+        raise ValueError("lower_value must be < upper_value.")
+
+    if max_delta_hue == 0:
+        delta_hue = 0
+    else:
+        delta_hue = tf.random.uniform([],
+                                      -max_delta_hue,
+                                      max_delta_hue,
+                                      seed=seed)
+    if lower_saturation == upper_saturation:
+        scale_saturation = lower_saturation
+    else:
+        scale_saturation = tf.random.uniform([],
+                                             lower_saturation,
+                                             upper_saturation,
+                                             seed=seed)
+    if lower_value == upper_value:
+        scale_value = lower_value
+    else:
+        scale_value = tf.random.uniform([],
+                                        lower_value,
+                                        upper_value,
+                                        seed=seed)
+    return adjust_hsv_in_yiq(image, delta_hue, scale_saturation, scale_value)
+
+
+@tf.function
+def adjust_hsv_in_yiq(image,
+                      delta_hue=0,
+                      scale_saturation=1,
+                      scale_value=1,
+                      name="adjust_hsv_in_yiq"):
+    """Adjust hue, saturation, value of an RGB image in YIQ color space.
+
+    This is a convenience method that converts an RGB image to float
+    representation, converts it to YIQ, rotates the color around the
+    Y channel by delta_hue in radians, scales the chrominance channels
+    (I, Q) by scale_saturation, scales all channels (Y, I, Q) by scale_value,
+    converts back to RGB, and then back to the original data type.
+
+    `image` is an RGB image.  The image hue is adjusted by converting the
+    image to YIQ, rotating around the luminance channel (Y) by
+    `delta_hue` in radians, multiplying the chrominance channels (I, Q)  by
+    `scale_saturation`, and multiplying all channels (Y, I, Q)  by
+    `scale_value`.  The image is then converted back to RGB.
+
+    Args:
+      image: RGB image or images. Size of the last dimension must be 3.
+      delta_hue: float, the hue rotation amount, in radians.
+      scale_saturation: float, factor to multiply the saturation by.
+      scale_value: float, factor to multiply the value by.
+      name: A name for this operation (optional).
+
+    Returns:
+      Adjusted image(s), same shape and DType as `image`.
+    """
+    with tf.name_scope(name):
```

Comment:
I prefer `with tf.name_scope(name, 'adjust_hsv_in_yiq', [image]) as name:`, what do you think?

Reply:
Umm, another annoying thing here is the doc for `name_scope` (seems wrong). The actual `name_scope` in TF2 can only feed `name` into the constructor.

https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/framework/ops.py#L6366

Reply:
![image](https://user-images.githubusercontent.com/11615393/54122626-8b73f300-4438-11e9-87d7-b632777a1515.png)